exercises-python/mike_sullivan/ex01_basics/main.py

```diff
@@ -0,0 +1,71 @@
+'''
+Write a command line program that will:
+
+Take a filename of an ascii file as an argument (you can use the example file here)
+
+Load that ascii file.
+
+Count the number of occurrences of unique words (longer than 4 characters and split hyphenated words, treating each part as different words). It should be case and punctuation insensitive. You only need to consider the following punctuation characters .,?'"!(): (hint: you will need a backslash escape character for the double-quote)
+
+Consider handling of common error cases, such as the wrong file name specified. Return error and status information to the user of the command line tool.
+
+Print the results to screen showing the unique words and the number of uses in descending order of usage, e.g.
+
+
+'''
+
+import argparse
+import re
+import numpy as np
```

Comment:
In the checks I can see `ModuleNotFoundError: No module named 'numpy'` because the tests run in a fairly minimal python environment.

You could either try rewriting `getuniquewords()` to avoid using numpy. https://docs.python.org/3/library/collections.html?highlight=counter#collections.Counter might be useful. (Though in actual mantid dev work we use numpy a lot)

Or, you could adjust the test workflow to install numpy. There is a file `.github/workflows/python-tests.yml` that tells github actions what to do. A step to do `pip install numpy` would work.